hudi-common/src/main/java/org/apache/hudi/common/table/log/BaseHoodieMergedLogRecordScanner.java

```diff
@@ -87,6 +94,11 @@ protected BaseHoodieMergedLogRecordScanner(HoodieStorage storage, String basePat
       this.records = new ExternalSpillableMap<>(maxMemorySizeInBytes, spillableMapBasePath, new DefaultSizeEstimator(),
           new HoodieRecordSizeEstimator(readerSchema), diskMapType, isBitCaskDiskMapCompressionEnabled);
       this.scannedPrefixes = new HashSet<>();
+
+      // TODO: use reader context for both non fg reader.
+      this.orderingFieldName = Option.ofNullable(ConfigUtils.getOrderingField(payloadProps)).orElseGet(() -> hoodieTableMetaClient.getTableConfig().getPreCombineField());
+      this.orderingFieldType = AvroSchemaUtils.findNestedFieldType(readerSchema, this.orderingFieldName).orElse(Schema.Type.INT);
+      this.orderingFieldDefault = TypeCaster.castValue(0, orderingFieldType);
```

Comment:
Default should always be integer-typed 0 for backwards compatibility.

Reply:
Done.